### DISCUSSION TITLE
Is this what it looks like when 6N138 doesn't work?

### DISCUSSION THREAD

**Discussion by warpeggio** [probonopd/MiniDexed#281] [Q&A] [answered]
Here's a scope shot of my attempt at serial midi using 6N138. I've been over the schematic a dozen times now and i'm confident my wiring is correct. Minidexed only starts making noise if i mash the keybed and force a serial parsing glitch. USB Midi and MIDI via USB Adapter are working for my build, for what it's worth.
![20220611_143556](https://user-images.githubusercontent.com/74164317/173202524-d0aa3a81-b81a-4286-9c53-f37da448f204.jpg)
Blue channel is the signal on the midi cable sink pin. 
Yellow channel is the output of the optocoupler on pin 6.

**Comment by luisfcorreia** [comment]
Can you please tell us more about how you have it connected?
Specifically the resistors used on the output side

**Reply by warpeggio** [reply]
Sure! Followed the schematic on the wiki for Classic Midi. You think maybe i got the wrong ones on there?
I *meant* to build it as 220k between pins 7 & ground, and
3k3 from pin 6 to the 3v3 rail.

I used raspi pin 17 for 3v3 power, just a convenient place to solder alongside the power for the DAC.

**Comment by luisfcorreia** [comment, marked as answer]
That's odd...

If you have a 100K resistor, try using it instead of the 220K

the signal shape is ok on going low, but slow to return high

but yes, very odd

**Reply by warpeggio** [reply]
Thanks for looking with me, @luisfcorreia and confirming my suspicion that the signal looked slow to return high. I'll double check all my resistor values and then try the 100k in its place.

**Comment by luisfcorreia** [comment]
well, the circuit works for me on a Raspi 1b. that's why I find it odd

**Comment by warpeggio** [comment]
With the 100k resistor, performance is much better but not perfect - i still have to hit the panic button more than i'd care to. It's just about usable at "human operator" speeds, but if you feed it an arpeggiator, it falls apart
![20220611_152444](https://user-images.githubusercontent.com/74164317/173203866-ac50730b-b6a1-496c-a5f9-c05fc84033bd.jpg)
.

**Comment by warpeggio** [comment]
I guess i can try another 6N138 from this same batch, but, uh....
![20220611_111617](https://user-images.githubusercontent.com/74164317/173203921-8a6aa2b5-e3be-4bb9-bf25-b506805d0425.jpg)

**Reply by luisfcorreia** [reply]
(resistor on pin 6 does not seem to be probably soldered)

at least from this perspective

**Reply by warpeggio** [reply]
I thought the same thing after looking at this image, but i double checked and everything is connected (at least now 😄 )

**Comment by luisfcorreia** [comment]
How's the waveform now with a 100k resistor now?

**Reply by warpeggio** [reply]
See above - it works MUCH better, but with the occasional glitch still. I think your advice below about trying values as low as 50k is helpful. I agree with your assessment that this is probably due to chip variation.

**Comment by diyelectromusic** [comment]
100k/220k seems pretty high for that pin 7 to GND resistor based on other circuits I've seen... I've used a 4K7 with success, but then if using a 6N138 then I'd tend to power it from 5V even though pulling RX high to 3V3...

But I'm not an electronics person really, I'm just going on what I've read in places like:
https://electronics.stackexchange.com/questions/165255/midi-in-many-circuits
https://gr33nonline.wordpress.com/2019/05/15/opto-couple-confusion-6n137-or-6n138/
https://www.samplerbox.org/article/midiinwithrpi

And the two published designs for 3V3 MIDI IN circuits from the Adafruit Feather Wing and the published circuit for the Teensy.

It would seem various circuits have used values between 4K7 and 10K... but I've given up trying to fully understand why myself.  I'm guessing the higher the resistance, the lower the current so the slower the transition response?  But as I say, I'm not really an electronics person, I just dabble...

Kevin

**Reply by luisfcorreia** [reply]
the output resistor is there to make output go fast from hi to low, the lower that resistor is, the faster it will go...

the resistor for pin 7 is there to make transition from low to high go faster.
again, the lower it is, the faster it will go from low to high,
HOWEVER, if it is very low (about 50k or less) it will make it go slower from hi to low

I know all chips are not made all alike and one must test with several values, or even using potentiometers until you get a "response that fits"

**Reply by warpeggio** [reply]
It works well enough now with the 100k resistor that i spent several minutes exploring some .syx files i had loaded 😁 . I need to take a break for a while as i have been at this all day, but i want to try it with a ~55k resistor and see if that helps or hurts.

I see that the MT32-pi project has a schematic based around H11L1, so i've got that in my back pocket if i can't get this to perform right.

All in all though, i must say the sound quality and latency is absolutely delightful. Definitely a worthwhile project!!!

**Reply by luisfcorreia** [reply]
yes, other opto-couplers may be better at it
I have a 4 pin opto which is a LOT newer and needs only one resistor.

I promise to test it again, the next time I revisit this hardware build
(stuck on other stuff for now)

**Comment by warpeggio** [comment]
SUCCESS! With a 56k resistor for R3 of the schematic (the pulldown resistor between pins 7 and 5 / ground), Minidexed had zero glitches at max rate from my Keystep arpeggiator.
![20220611_184309](https://user-images.githubusercontent.com/74164317/173208421-29b14cee-f070-4364-aa77-e74a086d1933.jpg)
Thanks so much for your help, @luisfcorreia, and cheers to you as well @diyelectromusic for the helpful links.

**Comment by probonopd** [comment]
This community is awesome :+1:
